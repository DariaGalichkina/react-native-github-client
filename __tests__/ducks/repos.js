
import mockStore from 'redux-mock-store';

import reducer, { getRepos, setReposData } from '../../app/ducks/repos.js';

describe('Repos # reducer', () => {

  it('is expected to return appropriate state', () => {
    expect(reducer(null, setReposData([{ name: 'name' }]))).toMatchSnapshot();
  });

});

describe('Repos # async/getRepos', () => {

  afterEach(() => nock.cleanAll())

  it('is expected to trigger set data action with received data', async () => {

    nock('https://api.github.com')
      .get('/user/repos')
      .reply(200, [{ name: 'repo-name', stargazers_count: 10 }]);

    const store = mockStore({ repos: null });

    await store.dispatch(getRepos());

    expect(store.getActions()).toMatchSnapshot();

  });

});

describe('Repos # setReposData', () => {

  it('is expected to create SET_DATA action', () => {
    expect(setReposData([ { name: 'name', stargazers_count: 5 } ])).toMatchSnapshot();
  });

});
