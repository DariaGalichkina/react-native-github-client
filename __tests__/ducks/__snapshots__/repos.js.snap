// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos # async/getRepos is expected to trigger set data action with received data 1`] = `
Array [
  Object {
    "reposData": Array [
      Object {
        "name": "repo-name",
        "stargazers_count": 10,
      },
    ],
    "type": "repos/SET_DATA",
  },
]
`;

exports[`Repos # reducer is expected to return appropriate state 1`] = `
Array [
  Object {
    "name": "name",
  },
]
`;

exports[`Repos # setReposData is expected to create SET_DATA action 1`] = `
Object {
  "reposData": Array [
    Object {
      "name": "name",
      "stargazers_count": 5,
    },
  ],
  "type": "repos/SET_DATA",
}
`;
